lib/tpm2_eventlog_ascii.c

```diff
@@ -0,0 +1,208 @@
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <tss2/tss2_tpm2_types.h>
+
+#include "log.h"
+#include "tcg-efi-event.h"
+#include "tpm2_eventlog.h"
+#include "tpm2_eventlog_ascii.h"
+
+char* get_alg_name(UINT16 alg_id) {
+
+    switch (alg_id) {
+    case TPM2_ALG_SHA1:
+        return "TPM2_ALG_SHA1";
+    case TPM2_ALG_SHA256:
+        return "TPM2_ALG_SHA256";
+    case TPM2_ALG_SHA384:
+        return "TPM2_ALG_SHA384";
+    case TPM2_ALG_SHA512:
+        return "TPM2_ALG_SHA512";
+    default:
+        return "UNKNOWN_ALGORITHM";
+    }
+}
+char* eventtype_to_string (UINT32 event_type) {
```

Comment:
`const char *`

Also, here and throughout, `<type> *<varname>`. The * should be tight to the variable name not the type.

Reply:
Good catch. Went with `char const *`.